docs/docs/reference/intersection-types-spec.md

```diff
@@ -0,0 +1,104 @@
+---
+layout: doc-page
+title: "Intersection Types - More Details"
+---
+
+## Syntax
+
+Syntactically, an intersection type `S & T` is similar to an infix type, where
+the infix operator is `&`. `&` is treated as a soft keyword. That is, it is a
+normal identifier with the usual precedence. But a type of the form `A & B` is
+always recognized as an intersection type, without trying to resolve `&`.
+
+```
+Type              ::=  ...| InfixType
+InfixType         ::=  RefinedType {id [nl] RefinedType}
+```
+
+## Subtyping Rules
+
+```
+T <: A    T <: B
+----------------
+  T <: A & B
+
+    A <: T
+----------------
+    A & B <: T
+
+    B <: T
+----------------
+    A & B <: T
+```
+
+From the rules above, we can show that `&` is _commutative_: `A & B <: B & A` for any type `A` and `B`.
+
+```
+   B <: B           A <: A
+----------       -----------
+A & B <: B       A & B <: A
+---------------------------
+       A & B  <:  B & A
+```
+
+In another word, `A & B` is the same type as `B & A`, in the sense that the two types
+have the same values and are subtypes of each other.
+
+If `C` is a type constructor, the join `C[A] & C[B]` is simplified by pulling the
+intersection inside the constructor, using the following two rules:
+
+- If `C` is covariant, `C[A] & C[B] ~> C[A & B]`
+- If `C` is contravariant, `C[A] & C[B] ~> C[A | B]`
+
```

Comment:
Where does the "simplified" terminology and the `~>` come from? Intuitively it makes sense, but seems redundant with the typing rules stating the same thing immediately below.